RelayProofsOfConcept/EddsaTendermintServer/mmpc-client/src/bin/kg-client.rs

```diff
@@ -173,10 +185,51 @@ fn main() {
             {
                 session.handle_relay_message(msg.clone());
             }
-            return;
+            break;
         }
         let server_response = session.query();
         session.store_server_response(&server_response);
         thread::sleep(time::Duration::from_millis(RETRY_TIMEOUT));
     }
+    let total_time = start_time.elapsed().expect("Weird time");
+    println!("{:}", total_time.as_millis());
+
+    if let Err(err) = write_to_csv(client_index, total_time.as_millis() as u32, capacity) {
+        println!("error running example: {}", err);
+        process::exit(1);
+    }
+}
+
+fn write_to_csv(index: u32, millis: u32, capacity: u32) -> Result<(), Box<dyn Error>> {
```

Comment:
this might not be a good approach.
as  parties may run on different   instances/containers

will be better if we specify the   index in the file name?
then after we gather all the data from different parties   in to a same folder, we can perform  synthesis/statistics